src/responder/nss/nss_cmd.c

```diff
@@ -451,6 +451,61 @@ static errno_t nss_getby_sid(struct cli_ctx *cli_ctx,
     return EOK;
 }
 
+static errno_t nss_getby_addr(struct cli_ctx *cli_ctx,
+                              enum cache_req_type type,
+                              enum sss_mc_type memcache,
+                              nss_protocol_fill_packet_fn fill_fn)
+{
+    struct cache_req_data *data;
+    struct nss_cmd_ctx *cmd_ctx;
+    struct tevent_req *subreq;
+    uint8_t *addr;
+    uint32_t addrlen;
+    uint32_t af;
+    errno_t ret;
+
+    cmd_ctx = nss_cmd_ctx_create(cli_ctx, cli_ctx, type, fill_fn);
+    if (cmd_ctx == NULL) {
+        ret = ENOMEM;
+        goto done;
+    }
+
+    cmd_ctx->flags = 0;
+    ret = nss_protocol_parse_addr(cli_ctx, &af, &addrlen, &addr);
+    if (ret != EOK) {
+        DEBUG(SSSDBG_CRIT_FAILURE, "Failed to parse address: %s\n",
+	      strerror(ret));
+        goto done;
+    }
+
+    data = cache_req_data_addr(cmd_ctx, type, af, addrlen, addr);
+    if (data == NULL) {
+        DEBUG(SSSDBG_CRIT_FAILURE, "Failed to set cache request data!\n");
+        ret = ENOMEM;
+        goto done;
+    }
+
+    subreq = nss_get_object_send(cmd_ctx, cli_ctx->ev, cli_ctx,
+                                 data, SSS_MC_NONE, NULL, 0);
```

Comment:
I think it would be better to use the `memcache` parameter here instead of just `SSS_MC_NONE`.